dm/syncer/handle_error.go

```diff
@@ -29,47 +30,58 @@ import (
 )
 
 // HandleError handle error for syncer.
-func (s *Syncer) HandleError(ctx context.Context, req *pb.HandleWorkerErrorRequest) error {
+func (s *Syncer) HandleError(ctx context.Context, req *pb.HandleWorkerErrorRequest) (string, error) {
 	pos := req.BinlogPos
 
 	if len(pos) == 0 {
 		startLocation, isQueryEvent := s.getErrLocation()
 		if startLocation == nil {
-			return fmt.Errorf("source '%s' has no error", s.cfg.SourceID)
+			return "", fmt.Errorf("source '%s' has no error", s.cfg.SourceID)
 		}
+
 		if !isQueryEvent {
-			return fmt.Errorf("only support to handle ddl error currently, see https://docs.pingcap.com/tidb-data-migration/stable/error-handling for other errors")
+			return "", fmt.Errorf("only support to handle ddl error currently, see https://docs.pingcap.com/tidb-data-migration/stable/error-handling for other errors")
 		}
 		pos = startLocation.Position.String()
 	} else {
 		startLocation, err := binlog.VerifyBinlogPos(pos)
 		if err != nil {
-			return err
+			return "", err
 		}
 		pos = startLocation.String()
 	}
 
 	events := make([]*replication.BinlogEvent, 0)
 	var err error
-	if req.Op == pb.ErrorOp_Replace {
+	if req.Op == pb.ErrorOp_Replace || req.Op == pb.ErrorOp_Inject {
 		events, err = s.genEvents(ctx, req.Sqls)
 		if err != nil {
-			return err
+			return "", err
 		}
 	}
 
 	// remove outdated operators when add operator
 	err = s.errOperatorHolder.RemoveOutdated(s.checkpoint.FlushedGlobalPoint())
 	if err != nil {
-		return err
+		return "", err
+	}
+
+	if req.Op == pb.ErrorOp_List {
```

Comment:
the if branch of `ErrorOp_List` can be putat line 53

Reply:
ok